packages/core/core-flows/src/order/steps/register-shipment.ts

```diff
@@ -0,0 +1,29 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IOrderModuleService, RegisterOrderShipmentDTO } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type RegisterOrderShipmentStepInput = RegisterOrderShipmentDTO
+
+export const registerOrderShipmentStepId = "register-order-shipment"
+export const registerOrderShipmentStep = createStep(
+  registerOrderShipmentStepId,
+  async (data: RegisterOrderShipmentStepInput, { container }) => {
+    const service = container.resolve<IOrderModuleService>(
+      ModuleRegistrationName.ORDER
+    )
+
+    await service.registerShipment(data)
+    return new StepResponse(void 0, data.order_id)
+  },
+  async (orderId, { container }) => {
+    if (!orderId) {
+      return
+    }
+
+    const service = container.resolve<IOrderModuleService>(
+      ModuleRegistrationName.ORDER
+    )
+
+    await service.revertLastVersion(orderId)
```

Comment:
question: if another workflow is also updating the order, can it revert to the wrong last version? maybe it will be part of the locking mechanism right?

Reply:
Yeah, if there is concurrency we can revert the wrong change.
However this flow shouldn't be very concerning in regards concurrency. 
But I would love to have the locking as a Module so we can apply it to whatever flow we want to control this type of situation.

Reply:
But I will very likely change this to return a specific Order Change ID and try to delele only that. At least if something happen, the worst case scenario will fail to revert the change.